src/wp-admin/options-general.php

```diff
@@ -97,7 +97,77 @@
 <p class="description" id="tagline-description"><?php echo $tagline_description; ?></p></td>
 </tr>
 
+<tr class="hide-if-no-js">
+<th scope="row"><?php _e( 'Site Icon' ); ?></th>
+<td>
+	<?php
+
+	wp_enqueue_media();
+	wp_enqueue_script( 'site-icon' );
+
+	$classes_for_upload_button = 'upload-button button-add-media button-add-site-icon';
+	$classes_for_update_button = 'button';
+
+	$classes_for_avatar = 'avatar avatar-150';
+	if ( has_site_icon() ) {
+		$classes_for_avatar          .= ' has-site-icon';
+		$classes_for_button           = $classes_for_update_button;
+		$classes_for_button_on_change = $classes_for_upload_button;
+	} else {
+		$classes_for_avatar          .= ' hidden';
+		$classes_for_button           = $classes_for_upload_button;
+		$classes_for_button_on_change = $classes_for_update_button;
+	}
```

Comment:
```suggestion
	$classes_for_button           = 'upload-button button-add-media button-add-site-icon';
	$classes_for_button_on_change = 'button';

	$classes_for_avatar = 'avatar avatar-150';
	if ( has_site_icon() ) {
		$classes_for_avatar          .= ' has-site-icon';
	} else {
		$classes_for_avatar          .= ' hidden';
	}
```

Remove duplicate code.

Reply:
While the code looks like it could be a duplicate, it's actually different. Appyling this would cause the button to have the wrong appearance.